src/components/BaseComponents/BaseHeader.tsx

```diff
@@ -85,9 +108,19 @@ function BaseHeader(props: HeaderProps) {
       <div className={classes.toolbar}>
         <div className={classes.left}>{left}</div>
         <div className={classes.right}>{right}</div>
+        <Menu anchorEl={anchorEl} keepMounted open={Boolean(anchorEl)} onClose={closeMenu}>
+            <MenuItem>{name}</MenuItem>
+            <MenuItem>{email}</MenuItem>
+            <MenuItem onClick={handleLogoutClick}>Logout</MenuItem>
+        </Menu>
       </div>
     </div>
   );
 }
 
-export default withStyles(styles)(BaseHeader);
+const mapStateToProps = (state: RootState) => ({
+  name: state.userData.user?.fields.Name || '',
+  email: state.userData.user?.fields.Email || '',
```

Comment:
It'd probably be good to switch everything to getters at some point (ex: getUserName, getUserEmail, etc) for unified error handling and easier changes if we need it. We can do that in a future sweeping PR (lets not do it here). Just wanted to document the thought.